plugins/destination/gremlin/client/schema.json

```
@@ -0,0 +1,224 @@
+              },
+              "password": {
+                "type": "string",
+                "minLength": 1,
+                "description": "Password to connect to the database. Required when `auth_mode` is `basic`."
+              }
+            },
+            "required": [
+              "username",
+              "password"
+            ]
+          }
+        },
+        {
+          "if": {
+            "properties": {
+              "username": {
+                "type": "string",
+                "minLength": 1,
+                "description": "Username to connect to the database. Required when `auth_mode` is `basic`."
+              }
+            },
+            "required": [
+              "username"
+            ]
+          },
+          "then": {
+            "properties": {
+              "password": {
+                "type": "string",
+                "minLength": 1,
+                "description": "Password to connect to the database. Required when `auth_mode` is `basic`."
+              },
+              "auth_mode": {
+                "type": "string",
+                "enum": [
+                  "none",
+                  "basic",
+                  "aws"
+                ],
+                "const": "basic",
+                "description": "Authentication mode to use. `basic` uses static credentials, `aws` uses AWS IAM authentication.",
+                "default": "none"
+              }
+            },
+            "required": [
+              "password",
+              "auth_mode"
+            ]
+          }
+        },
+        {
+          "if": {
+            "properties": {
+              "password": {
+                "type": "string",
+                "minLength": 1,
+                "description": "Password to connect to the database. Required when `auth_mode` is `basic`."
+              }
+            },
+            "required": [
+              "password"
+            ]
+          },
+          "then": {
+            "properties": {
+              "username": {
+                "type": "string",
+                "minLength": 1,
+                "description": "Username to connect to the database. Required when `auth_mode` is `basic`."
+              },
+              "auth_mode": {
+                "type": "string",
+                "enum": [
+                  "none",
+                  "basic",
+                  "aws"
+                ],
+                "const": "basic",
+                "description": "Authentication mode to use. `basic` uses static credentials, `aws` uses AWS IAM authentication.",
+                "default": "none"
+              }
+            },
+            "required": [
+              "username",
+              "auth_mode"
+            ]
+          }
+        }
+      ],
+      "properties": {
+        "endpoint": {
+          "type": "string",
+          "pattern": "^ws(s?)://[^\n\\:]+(\\:[0-9]{1,5})?$",
```

no need to group, _ do we really need to check for port here? maybe just use a simple verification?
```suggestion
          "pattern": "^wss?\://[^\n]+$",
```

---

you want verification but not too much 😂